File: my-shelf-server/src/main/java/com/myshelf/wardrobe/dto/AuthResponse.java
package com.myshelf.wardrobe.dto;

/**
 * DTO для ответа при аутентификации и регистрации.
 */
public class AuthResponse {
    private String token;
    private UserProfileDTO profile;

    public AuthResponse() {
    }

    public AuthResponse(String token, UserProfileDTO profile) {
        this.token = token;
        this.profile = profile;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public UserProfileDTO getProfile() {
        return profile;
    }

    public void setProfile(UserProfileDTO profile) {
        this.profile = profile;
    }
}

File: myshelf-app/app/src/main/java/com/myshelf/myshelf_app/data/remote/dto/AuthResponse.kt
package com.myshelf.myshelf_app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    @SerializedName("token")
    val token: String,

    @SerializedName("profile")
    val profile: UserProfileResponse? = null
) {
    /** Идентификатор пользователя из профиля (совместимость с серверным AuthResponse). */
    val userId: String?
        get() = profile?.id
}
